api/image_pull.go

```diff
@@ -11,10 +11,17 @@ import (
 )
 
 // ImagePull pulls a image from a remote location to local storage
-func (c *API) ImagePull(ctx context.Context, nameWithTag string) (string, error) {
+func (c *API) ImagePull(ctx context.Context, nameWithTag string, auth ImageAuthConfig) (string, error) {
 	var id string
 
-	res, err := c.Post(ctx, fmt.Sprintf("/v1.0.0/libpod/images/pull?reference=%s", nameWithTag), nil)
+	authHeader, err := NewAuthHeader(auth)
```

Comment:
I think we may want to check if the supplied `ImageAuthConfig` is the zero value or use a pointer, so we can check if auth is empty and conditionally set it.

If ImageAuthConfig is an empty struct right now I think it will get json encoded to `{}` and then base64 encoded to `e30=`, which may be confusing or potentially result in an error. https://play.golang.org/p/xHknDDMTl8A

Reply:
done